lib/Command/FixEncryptedVersion.php

```diff
@@ -0,0 +1,204 @@
+
+	/** @var View  */
+	private $view;
+
+	public function __construct(IRootFolder $rootFolder, IUserManager $userManager, View $view) {
+		$this->rootFolder = $rootFolder;
+		$this->userManager = $userManager;
+		$this->view = $view;
+		parent::__construct();
+	}
+
+	protected function configure() {
+		parent::configure();
+
+		$this
+			->setName('encryption:fixencryptedversion')
+			->setDescription('Fix the encrypted version if the encrypted file(s) are not downloadable from the UI')
+			->addArgument(
+				'user',
+				InputArgument::REQUIRED,
+				'The user id whose files needs fix'
+			);
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$user = $input->getArgument('user');
+
+		if ($user === null) {
+			$output->writeln("<error>No user provided.</error>\n");
+		}
+
+		if ($this->userManager->get($user) === null) {
+			$output->writeln("<error>User $user does not exist. Kindly provide a valid user id</error>");
+			return 1;
+		}
+		$this->walkUserFolder($user, $output);
+	}
+
+	/**
+	 * @param string $user
+	 * @param OutputInterface $output
+	 */
+	private function walkUserFolder($user, OutputInterface $output) {
+		$this->setupUserFs($user);
+		$directories = [];
+		$directories[] = '/' . $user . '/files';
+		while ($root = \array_pop($directories)) {
+			$directoryContent = $this->view->getDirectoryContent($root);
+			foreach ($directoryContent as $file) {
+				$path = $root . '/' . $file['name'];
+				if ($this->view->is_dir($path)) {
+					$directories[] = $path;
+				} else {
+					$output->writeln("Verifying the content of file $path");
+					$this->verifyFileContent($path, $output);
+				}
+			}
+		}
+	}
+
+	/**
+	 * @param string $path
+	 * @param OutputInterface $output
+	 * @param bool $allowOnce
+	 */
+	private function verifyFileContent($path, OutputInterface $output, $allowOnce = true) {
+		try {
+			if ($this->view->readfilePart($path, 0, 9000) !== false) {
+				$output->writeln("<info>The file $path is: OK</info>");
+			}
+			return true;
+		} catch (HintException $e) {
+			\OC::$server->getLogger()->warning("Issue: " . $e->getMessage());
+			//If allowOnce is set to false, this becomes recursive.
+			if ($allowOnce === true) {
+				//Lets rectify the file by correcting encrypted version
+				return $this->correctEncryptedVersion($path, $output);
+			}
+			return false;
+		}
+	}
+
+	/**
+	 * @param string $path
+	 * @param OutputInterface $output
+	 */
+	private function correctEncryptedVersion($path, OutputInterface $output) {
+		$skipDecrymentedVersion = null;
+
+		$fixedFileInfo = ['filename' => '', 'encryptedVersion' => 0];
+
+		$fileInfo = $this->view->getFileInfo($path);
+		$fileId = $fileInfo->getId();
+		$encryptedVersion = $fileInfo->getEncryptedVersion();
+		$oldEncryptedVersion = $encryptedVersion;
+
+		$storage = $fileInfo->getStorage();
+
+		$cache = new Cache($storage);
+		$fileCache = $cache->get($fileId);
+		$fileCacheStorage = $fileCache->getData()['storage'];
```

Comment:
No more relying on the string. Because in the updated version of pr I have used :
```
if ($storage->instanceOfStorage('OCA\Files_Sharing\ISharedStorage'))
```